<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PhaseVis</class>
 <widget class="QMainWindow" name="PhaseVis">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1152</width>
    <height>801</height>
   </rect>
  </property>
  <property name="maximumSize">
   <size>
    <width>3000</width>
    <height>3000</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>PhaseVis</string>
  </property>
  <property name="locale">
   <locale language="English" country="UnitedStates"/>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_2">
    <item>
     <widget class="GLWidget" name="Display" native="true">
      <property name="minimumSize">
       <size>
        <width>480</width>
        <height>480</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>2000</width>
        <height>2000</height>
       </size>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QMenuBar" name="MenuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1152</width>
     <height>20</height>
    </rect>
   </property>
   <property name="defaultUp">
    <bool>false</bool>
   </property>
   <property name="nativeMenuBar">
    <bool>false</bool>
   </property>
   <widget class="QMenu" name="menuMain">
    <property name="title">
     <string>Main</string>
    </property>
    <addaction name="saveFrameBufferAction"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuAbout">
    <property name="title">
     <string>About</string>
    </property>
    <addaction name="actionAuthor"/>
    <addaction name="actionApplication"/>
    <addaction name="actionModels"/>
   </widget>
   <widget class="QMenu" name="menuActions">
    <property name="title">
     <string>Vis</string>
    </property>
    <addaction name="importGridAction"/>
    <addaction name="separator"/>
    <addaction name="addMethodAsChildAction"/>
    <addaction name="addMethodAsRowAction"/>
    <addaction name="separator"/>
    <addaction name="removeMethodAction"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="actionVisualization_Pipeline"/>
    <addaction name="actionVisualization_Options"/>
    <addaction name="actionDisplay_Options"/>
    <addaction name="actionLayer_Manager"/>
   </widget>
   <widget class="QMenu" name="menuGenerate_Data">
    <property name="title">
     <string>Generate Data</string>
    </property>
    <addaction name="actionVanDerPol"/>
    <addaction name="actionVanDerPolDuffing"/>
    <addaction name="actionMorrisLecar"/>
    <addaction name="actionAndronovHopf"/>
   </widget>
   <addaction name="menuMain"/>
   <addaction name="menuActions"/>
   <addaction name="menuGenerate_Data"/>
   <addaction name="menuView"/>
   <addaction name="menuAbout"/>
  </widget>
  <widget class="QDockWidget" name="dockVisPipeline">
   <property name="features">
    <set>QDockWidget::AllDockWidgetFeatures</set>
   </property>
   <property name="windowTitle">
    <string notr="true">Visualization Pipeline</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QTreeView" name="treePipeline">
       <property name="alternatingRowColors">
        <bool>true</bool>
       </property>
       <attribute name="headerVisible">
        <bool>false</bool>
       </attribute>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockPipelineOptions">
   <property name="windowTitle">
    <string>Visualization Options</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_3">
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QTableView" name="optionsView">
       <property name="dragEnabled">
        <bool>false</bool>
       </property>
       <property name="dragDropMode">
        <enum>QAbstractItemView::NoDragDrop</enum>
       </property>
       <property name="alternatingRowColors">
        <bool>true</bool>
       </property>
       <property name="selectionMode">
        <enum>QAbstractItemView::SingleSelection</enum>
       </property>
       <property name="selectionBehavior">
        <enum>QAbstractItemView::SelectRows</enum>
       </property>
       <property name="showGrid">
        <bool>true</bool>
       </property>
       <attribute name="horizontalHeaderCascadingSectionResizes">
        <bool>false</bool>
       </attribute>
       <attribute name="horizontalHeaderStretchLastSection">
        <bool>true</bool>
       </attribute>
       <attribute name="verticalHeaderVisible">
        <bool>false</bool>
       </attribute>
       <attribute name="verticalHeaderStretchLastSection">
        <bool>false</bool>
       </attribute>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_8">
       <item>
        <widget class="QPushButton" name="executeButton">
         <property name="maximumSize">
          <size>
           <width>100</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string>Execute</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="addToLayersButton">
         <property name="maximumSize">
          <size>
           <width>100</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string>Add to Layers</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockDisplayOptions">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="allowedAreas">
    <set>Qt::LeftDockWidgetArea|Qt::RightDockWidgetArea</set>
   </property>
   <property name="windowTitle">
    <string>Display Options</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_4">
    <layout class="QVBoxLayout" name="verticalLayout_4">
     <item>
      <widget class="QTreeWidget" name="treeWidget_2">
       <property name="alternatingRowColors">
        <bool>true</bool>
       </property>
       <column>
        <property name="text">
         <string notr="true">Option</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Value</string>
        </property>
       </column>
       <item>
        <property name="text">
         <string>Lighting Options</string>
        </property>
        <item>
         <property name="text">
          <string>Light Position</string>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="flags">
          <set>ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
         </property>
         <item>
          <property name="text">
           <string>x</string>
          </property>
          <property name="text">
           <string>0</string>
          </property>
          <property name="flags">
           <set>ItemIsSelectable|ItemIsEditable|ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
          </property>
         </item>
         <item>
          <property name="text">
           <string>y</string>
          </property>
          <property name="text">
           <string>0</string>
          </property>
          <property name="flags">
           <set>ItemIsSelectable|ItemIsEditable|ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
          </property>
         </item>
         <item>
          <property name="text">
           <string>z</string>
          </property>
          <property name="text">
           <string>0</string>
          </property>
          <property name="flags">
           <set>ItemIsSelectable|ItemIsEditable|ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
          </property>
         </item>
         <item>
          <property name="text">
           <string>w</string>
          </property>
          <property name="text">
           <string>0</string>
          </property>
          <property name="flags">
           <set>ItemIsSelectable|ItemIsEditable|ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
          </property>
         </item>
        </item>
        <item>
         <property name="text">
          <string>Ambient Light</string>
         </property>
         <item>
          <property name="text">
           <string>R</string>
          </property>
          <property name="text">
           <string>0</string>
          </property>
          <property name="flags">
           <set>ItemIsSelectable|ItemIsEditable|ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
          </property>
         </item>
         <item>
          <property name="text">
           <string>G</string>
          </property>
          <property name="text">
           <string>0</string>
          </property>
          <property name="flags">
           <set>ItemIsSelectable|ItemIsEditable|ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
          </property>
         </item>
         <item>
          <property name="text">
           <string>B</string>
          </property>
          <property name="text">
           <string>0</string>
          </property>
          <property name="flags">
           <set>ItemIsSelectable|ItemIsEditable|ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
          </property>
         </item>
         <item>
          <property name="text">
           <string>A</string>
          </property>
          <property name="text">
           <string>1</string>
          </property>
          <property name="flags">
           <set>ItemIsSelectable|ItemIsEditable|ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
          </property>
         </item>
        </item>
        <item>
         <property name="text">
          <string>Diffuse Light</string>
         </property>
         <item>
          <property name="text">
           <string>R</string>
          </property>
          <property name="text">
           <string>0</string>
          </property>
          <property name="flags">
           <set>ItemIsSelectable|ItemIsEditable|ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
          </property>
         </item>
         <item>
          <property name="text">
           <string>G</string>
          </property>
          <property name="text">
           <string>0</string>
          </property>
          <property name="flags">
           <set>ItemIsSelectable|ItemIsEditable|ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
          </property>
         </item>
         <item>
          <property name="text">
           <string>B</string>
          </property>
          <property name="text">
           <string>0</string>
          </property>
          <property name="flags">
           <set>ItemIsSelectable|ItemIsEditable|ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
          </property>
         </item>
         <item>
          <property name="text">
           <string>A</string>
          </property>
          <property name="text">
           <string>1</string>
          </property>
          <property name="flags">
           <set>ItemIsSelectable|ItemIsEditable|ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
          </property>
         </item>
        </item>
       </item>
       <item>
        <property name="text">
         <string>Orientation &amp; scaling</string>
        </property>
        <item>
         <property name="text">
          <string>Rotation</string>
         </property>
         <item>
          <property name="text">
           <string>X rotatiom</string>
          </property>
          <property name="flags">
           <set>ItemIsSelectable|ItemIsEditable|ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Y rotation</string>
          </property>
          <property name="flags">
           <set>ItemIsSelectable|ItemIsEditable|ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Z rotation</string>
          </property>
          <property name="flags">
           <set>ItemIsSelectable|ItemIsEditable|ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
          </property>
         </item>
        </item>
        <item>
         <property name="text">
          <string>Scale</string>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="flags">
          <set>ItemIsSelectable|ItemIsEditable|ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
         </property>
        </item>
       </item>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="saveScreenshotButton">
       <property name="text">
        <string>Take screenshot</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockLayer_Manager">
   <property name="minimumSize">
    <size>
     <width>226</width>
     <height>150</height>
    </size>
   </property>
   <property name="windowTitle">
    <string>Layers Options</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_5">
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <widget class="QTableView" name="layerView">
       <property name="alternatingRowColors">
        <bool>true</bool>
       </property>
       <property name="selectionMode">
        <enum>QAbstractItemView::SingleSelection</enum>
       </property>
       <property name="selectionBehavior">
        <enum>QAbstractItemView::SelectRows</enum>
       </property>
       <property name="showGrid">
        <bool>false</bool>
       </property>
       <attribute name="horizontalHeaderStretchLastSection">
        <bool>true</bool>
       </attribute>
       <attribute name="verticalHeaderVisible">
        <bool>false</bool>
       </attribute>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QPushButton" name="drawButton">
         <property name="text">
          <string>Draw</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="removeLayerButton">
         <property name="text">
          <string>Remove from layer</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionAuthor">
   <property name="text">
    <string>Author</string>
   </property>
   <property name="toolTip">
    <string>Author</string>
   </property>
  </action>
  <action name="addMethodAsChildAction">
   <property name="text">
    <string>Add Method as Child</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="removeMethodAction">
   <property name="text">
    <string>Remove Method</string>
   </property>
  </action>
  <action name="actionVisualization_Pipeline">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Visualization Pipeline</string>
   </property>
  </action>
  <action name="actionVisualization_Options">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Visualization Options</string>
   </property>
  </action>
  <action name="actionDisplay_Options">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Display Options</string>
   </property>
  </action>
  <action name="importGridAction">
   <property name="text">
    <string>Import Grid</string>
   </property>
  </action>
  <action name="actionApplication">
   <property name="text">
    <string>Application</string>
   </property>
  </action>
  <action name="addMethodAsRowAction">
   <property name="text">
    <string>Add Method as Row</string>
   </property>
  </action>
  <action name="saveFrameBufferAction">
   <property name="text">
    <string>Save Frame Buffer as...</string>
   </property>
  </action>
  <action name="actionVanDerPol">
   <property name="text">
    <string>Van Der Pol Oscillator</string>
   </property>
  </action>
  <action name="actionVanDerPolDuffing">
   <property name="text">
    <string>Van Der Pol - Duffing Oscillator</string>
   </property>
  </action>
  <action name="actionMorrisLecar">
   <property name="text">
    <string>Morris - Lecar Model</string>
   </property>
  </action>
  <action name="actionModels">
   <property name="text">
    <string>Models</string>
   </property>
  </action>
  <action name="actionLayer_Manager">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Layer Manager</string>
   </property>
  </action>
  <action name="actionAndronovHopf">
   <property name="text">
    <string>Andronov - Hopf Oscillator</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>GLWidget</class>
   <extends>QWidget</extends>
   <header>Display/glwidget.h</header>
   <container>1</container>
   <slots>
    <signal>xRotationChanged(int)</signal>
    <signal>yRotationChanged(int)</signal>
    <signal>zRotationChanged(int)</signal>
    <signal>scaleChanged(double)</signal>
    <signal>listsBuilt()</signal>
    <slot>setXRotation(int)</slot>
    <slot>setYRotation(int)</slot>
    <slot>setZRotation(int)</slot>
    <slot>setScale(double)</slot>
    <slot>buildLists(UniformGrid*)</slot>
    <slot>saveImage()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>PhaseVis</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>541</x>
     <y>400</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionVisualization_Pipeline</sender>
   <signal>triggered(bool)</signal>
   <receiver>dockVisPipeline</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>67</x>
     <y>155</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dockVisPipeline</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>actionVisualization_Pipeline</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>67</x>
     <y>155</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionVisualization_Options</sender>
   <signal>triggered(bool)</signal>
   <receiver>dockPipelineOptions</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>67</x>
     <y>479</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dockPipelineOptions</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>actionVisualization_Options</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>67</x>
     <y>479</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionDisplay_Options</sender>
   <signal>triggered(bool)</signal>
   <receiver>dockDisplayOptions</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>889</x>
     <y>234</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dockDisplayOptions</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>actionDisplay_Options</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>889</x>
     <y>234</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>saveScreenshotButton</sender>
   <signal>clicked()</signal>
   <receiver>Display</receiver>
   <slot>saveImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>928</x>
     <y>204</y>
    </hint>
    <hint type="destinationlabel">
     <x>794</x>
     <y>639</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>saveFrameBufferAction</sender>
   <signal>triggered()</signal>
   <receiver>Display</receiver>
   <slot>saveImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>575</x>
     <y>400</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLayer_Manager</sender>
   <signal>triggered(bool)</signal>
   <receiver>dockLayer_Manager</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>1014</x>
     <y>591</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dockLayer_Manager</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>actionLayer_Manager</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1014</x>
     <y>591</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
